common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js

```diff
@@ -258,15 +268,23 @@ var dispatcher = {
 
     leaveOut : function(event) {
         this.out(event);
-        if(!this.contains(event.target, event.relatedTarget)) {
-            this.leave(event);
-        }
+        this.enterLeave(event, this.leave);
     },
 
     enterOver : function(event) {
         this.over(event);
-        if(!this.contains(event.target, event.relatedTarget)) {
-            this.enter(event);
+        this.enterLeave(event, this.enter);
+    },
+
+    enterLeave : function(event, fn) {
+        var target = event.target,
+            relatedTarget = event.relatedTarget;
+
+        while(target && !this.contains(target, relatedTarget)) {
```

Comment:
Hm, yep, haven't thought about it :( 

What if I rewrite it like this:

~~~js
if(!this.contains(target, relatedTarget)) {
  do {
    event.target = target;
    fn.call(this, event);  // we need to emit non bubbling events here on the subtree from target to relatedTarget
  } while(target = target.parentNode && target !== relatedTarget)
}
~~~

?